0101-nix formatting.md

```diff
@@ -0,0 +1,65 @@
+---
+feature: nix_formatting
+start-date: 2021-08-17
+author: Raphael Megzari
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+
+[summary]: #summary
+
+Decide on a recommended automated formatter for nix files in nixpkgs.
+
+# Motivation
+
+[motivation]: #motivation
+
+Prevent debate around how things should be formatted.
+
+# Detailed design
+
+[design]: #detailed-design
+
+The implementation of this RFC should include several parts
+
+- Agree on an automated formatter. [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt) is proposed here.
+- If adopted, agree on a schedule to format nixpkgs codebase. Should it be done at once, or on a per package basis.
+- Potentially agree on a hook to enforce formatting
+
+# Examples and Interactions
+
+[examples-and-interactions]: #examples-and-interactions
+
+This section is not needed for this RFC.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+- Having a commit that changes the formatting, can make git blame harder to use. It will need `--ignore-rev` flag.
+- Every formatter will have bugs. An example of a bug for nixpkgs-fmt can be found [here](https://github.com/NixOS/nixpkgs/pull/129392)
+
+# Alternatives
+
+[alternatives]: #alternatives
+
+- Alternative formatter [nixfmt](https://github.com/serokell/nixfmt)
```

Comment:
Enforces short line lengths which is not ideal with src in fetchers.

Reply:
If single string literal is too long, it is kept, so I see little problem with it. Can you provide example when it generates outright ugly results?